src/utils/api.js

```diff
@@ -10,3 +10,41 @@ export const getArticles = () => {
       return err;
     });
 };
+
+export const viewSingleArticle = (id) => {
+  let str = `https://portfolio-web-service-n7kk.onrender.com/api/articles`;
+
+  if (id) {
+    str += `/${id}`;
+    return axios
+      .get(str)
+      .then((response) => {
+        return response.data.article[0];
+      })
+      .catch((err) => {
+        return err;
+      });
+  } else {
+    return axios
+      .get(str)
+      .then((response) => {
+        return response.data.articles;
+      })
+      .catch((err) => {
+        return err;
+      });
+  }
+};
```

Comment:
I don't think you need this `else` block, the purpose of this function assumes the user will always provide an id, otherwise they should be using your previously created `getArticles` function.

Also you've named this function `viewSingleArticle` - where as the other two function names start with `get`. Consider making this naming more consistent, `getArticleById` would fit better.